core-bundle/src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -971,6 +971,77 @@ public static function ampersand($strString, $blnEncode=true): string
 	{
 		return preg_replace('/&(amp;)?/i', ($blnEncode ? '&amp;' : '&'), $strString);
 	}
+
+	/**
+	 * Converts an input-encoded string back to its raw UTF-8 value it originated from.
+	 *
+	 * It handles all Contao input encoding specifics like insert tags, basic entities and encoded entities.
+	 *
+	 * @param bool $blnRemoveInsertTags True to remove insert tags instead of replacing them
+	 */
+	public static function getRawDecodedValue(string $strValue, bool $blnRemoveInsertTags = false): string
```

Comment:
This method name needs to be changed. First of all, the `StringUtil` class does not have **any** `get…` method, because you're not _getting_ anything from it. Secondly, its unclear what _raw decoded_ means, and since this is the `StringUtil` class, what's _value_ but a string?.

Here some suggestions
 1. `revertInputEncoding`
 2. `decodeInput` / `restoreInput`
 3. `convertToPlainText` (which mean the other method would be `convertToPlainHtml`)

Reply:
If we go with `convertToPlainText` the second method should be named `convertToPlainTextFromHtml` IMO. WDYT?

Reply:
It's not "plain text" though. It's raw. Not plain... :)

Reply:
`convertToRawText()` and `convertToRawTextFromHtml()` or 
`decodeToRawText()` and `decodeToRawTextFromHtml()` ?

Reply:
I'm fine with either of them, yes :) @aschempp ?

Reply:
@Toflar Do you refer to `raw` from Twig? Then that would make sense. Otherwise to me, plain text means no encoding, no HTML etc., just what you would write in a (plain text) email.

In almost any case, running either of the method results in the same output, right? Expect the newline stuff for HTML, which is not "obvious" by "...FromHtml`. Maybe we can just have one method with an argument to `$blnConvertHtmlToNewline` ?

Reply:
No, I refer to English - the language. "Plain" means not encrypted to me.

Reply:
👎 for more arguments. Methods are better for that.

Reply:
`$blnConvertHtmlToNewline` is very confusing IMO, `decodeToRawTextFromHtml()` with the current method description should be way better understandable I think.

Reply:
https://en.wikipedia.org/wiki/Plain_text 🤷 

a better name for the argument would be `$preserveHtmlNewlines`. If looking at both method names, I would assume I must use the `FromHtml` version if my text contains HTML, but that wouldn't be correct.

Reply:
Yeah can we FINALLY just do something?

Reply:
> I would assume I must use the `FromHtml` version if my text contains HTML, but that wouldn't be correct.

That would be correct.

> Yeah can we FINALLY just do something?

I’m happy with any name for the two methods.

Reply:
> > I would assume I must use the FromHtml version if my text contains HTML, but that wouldn't be correct.

> That would be correct.

No it would not? It would only be correct if I want to keep newlines. Which for a page title I do not want. Both methods strip the HTML?

Reply:
We want to name the methods `revertInputEncoding()` and `htmlToPlainText()`.